sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,434 @@
+                f"sklearn.linear_model.{self.__class__.__name__}.fit"
+            )
+
+            if method_name == "fit":
+                return self._onedal_fit_supported(patching_status, method_name, *data)
+
+            if method_name in ["predict", "score"]:
+                return self._onedal_predict_supported(patching_status, method_name, *data)
+
+            raise RuntimeError(
+                f"Unknown method {method_name} in {self.__class__.__name__}"
+            )
+
+        def _initialize_onedal_estimator(self):
+            onedal_params = {
+                "fit_intercept": self.fit_intercept,
+                "alpha": self.alpha,
+                "copy_X": self.copy_X,
+            }
+            self._onedal_estimator = onedal_RidgeRegression(**onedal_params)
+
+        def _daal_fit(self, X, y, sample_weight=None):
+            daal4py_fit_ridge(self, X, y, sample_weight)
+            self._onedal_estimator.n_features_in_ = _num_features(X, fallback_1d=True)
+            self._onedal_estimator.coef_ = self.coef_
+            self._onedal_estimator.intercept_ = self.intercept_
+
+        def _onedal_fit(self, X, y, sample_weight, queue=None):
+            assert sample_weight is None
+
+            check_params = {
+                "X": X,
+                "y": y,
+                "dtype": [np.float64, np.float32],
+                "accept_sparse": ["csr", "csc", "coo"],
+                "y_numeric": True,
+                "multi_output": True,
+                "force_all_finite": False,
+            }
+            if sklearn_check_version("1.2"):
+                X, y = self._validate_data(**check_params)
+                self._validate_params()
+            else:
+                X, y = check_X_y(**check_params)
+
+            if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+                self._normalize = _deprecate_normalize(
+                    self.normalize,
+                    default=False,
+                    estimator_name=self.__class__.__name__,
+                )
+
+            self._initialize_onedal_estimator()
+            try:
+                # Falling back to daal4py if the device is CPU since
+                # onedal does not support non-scalars for alpha, thus
+                # should only be used for GPU/CPU with scalar alpha to not limit the functionality
+                cpu_device = queue is None or queue.sycl_device.is_cpu
+                if cpu_device and not _is_numeric_scalar(self.alpha):
+                    self._daal_fit(X, y)
+                else:
+                    self._onedal_estimator.fit(X, y, queue=queue)
+                self._save_attributes()
+
+            except RuntimeError:
+                logging.getLogger("sklearnex").info(
+                    f"{self.__class__.__name__}.fit "
+                    + get_patch_message("sklearn_after_onedal")
+                )
+
+                del self._onedal_estimator
+                super().fit(X, y)
```

Comment:
I have a strong opinion about this kind of fallbacks.

It should be removed. If something doesn't work on our side according to the given parameters, then it should be determined at the moment of dispatch call and within the framework of _onedal_cpu_support/_onedal_gpu_support checks.

I suggest remove this at all and try to address fails on `_onedal_cpu_support`/`_onedal_gpu_support` patching checks. I would like see how much test fails do we have in this case and want to see conformance test report.

Reply:
I agree, thanks for emphasizing this, inherited it from other algos as well. Will try your suggestion and report back.